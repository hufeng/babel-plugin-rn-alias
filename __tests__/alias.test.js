const alias = require('../lib');
const babel = require('babel-core');

const example = `
import React from 'react';
import ReactNative from 'react-native';
import * as plume2 from 'plume2'
`;

it('origin code', () => {
  expect(example).toMatchSnapshot();
});

it('match-snapshot', () => {
  const { code } = babel.transform(example, {
    plugins: [
      [
        alias,
        {
          alias: {
            react: 'qm-react',
            'react-native': 'qm-react-native',
            plume2: 'qm-plume2'
          }
        }
      ]
    ]
  });

  expect(code).toMatchSnapshot();
});
