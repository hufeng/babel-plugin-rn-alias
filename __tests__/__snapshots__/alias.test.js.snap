// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match-snapshot 1`] = `
"
import React from 'qm-react';
import ReactNative from 'qm-react-native';
import * as plume2 from 'qm-plume2';"
`;

exports[`origin code 1`] = `
"
import React from 'react';
import ReactNative from 'react-native';
import * as plume2 from 'plume2'
"
`;
